src/components/parameters-picker.tsx

```diff
@@ -18,39 +19,95 @@ export type ParametersPickerProps = {
   removeParameter: (idx: number) => void;
   // CSS classes for elements
   formPrefix: string;
+  errors: Scheduler.ErrorsType;
+  handleErrorsChange: (errors: Scheduler.ErrorsType) => void;
+};
+
+const useForceUpdate = () => {
+  const [, setTick] = useState(0);
+  const update = useCallback(() => {
+    setTick(tick => tick + 1);
+  }, []);
+  return update;
 };
 
 export function ParametersPicker(props: ParametersPickerProps): JSX.Element {
   const trans = useTranslator('jupyterlab');
 
+  const update = useForceUpdate();
+
+  const checkParameter = (
+    e: EventTarget & (HTMLInputElement | HTMLTextAreaElement)
+  ) => {
+    const paramInputName = e.name;
+
+    const paramMatch = paramInputName.match(/^parameter-(\d+)/);
+    if (paramMatch === null) {
+      return; // Invalid parameter name; should not happen
+    }
+    const paramIdx = parseInt(paramMatch[1]);
+
+    const modifiedParam = props.value[paramIdx];
+    const paramName = modifiedParam.name;
+    const paramValue = modifiedParam.value;
+
+    if (paramName === '' && paramValue !== '') {
+      const newErrors: Scheduler.ErrorsType = props.errors;
+      newErrors['parameter-' + paramIdx + '-name'] = trans.__(
+        'No name specified for this parameter.'
+      );
+      props.handleErrorsChange(newErrors);
+    } else {
+      const newErrors: Scheduler.ErrorsType = props.errors;
+      newErrors['parameter-' + paramIdx + '-name'] = '';
+      props.handleErrorsChange(newErrors);
+    }
+
+    // Force a rerender.
+    update();
```

Comment:
The reason you needed to force a re-render here is that React detects state changes based on object identity. Here you're taking `props.errors` and adding/modifying a property, rather than creating a new `errors` object.

Reply:
Shouldn't `handleErrorsChange`, aka `setErrors`, be modifying the state based on the modified/added property?

Reply:
Nope, it seems to use `Object.is()` to determine whether to trigger an update to the VDOM. Now, I tried finding documentation for this and I honestly can't find anything official; just a bunch of SO answers. Consider this secret FEE lore. 😅